core/schema/service.go

```diff
@@ -105,14 +123,87 @@ func (s *Service) Create(ctx context.Context, nsName string, schemaName string,
 		Compatibility: compatibility,
 	}
 	versionID := getIDforSchema(nsName, schemaName, sf.ID)
+	_, prevSchemaData, _ := s.GetLatest(ctx, nsName, schemaName)
 	version, err := s.repo.Create(ctx, nsName, schemaName, mergedMetadata, versionID, sf)
```

Comment:
error is ignored! Please handle it.

Reply:
Maybe report it with log line and a metric and skip the ChangeIdentification, or else the next flows might break

Reply:
even if don't handle it will be handled in 
```
if request.OldData == nil || len(request.OldData) == 0 {
		log.Println("IdentifySchemaChange failed as previous schema data is nil")
		return nil, errors.New("previous schema data is nil")
	}
```
Still added a if check.

Reply:
Don't add it just to address the comment. The common contract in golang is that if an error is returned by a method/function, that something the caller has to check for it and handle it.

What if the api provider i.e. Create here starts sending `emptyObject, nil` instead of `nil,nil` this might break the api users implementation.

Reply:
if you check the current implementation its returning nil and I think the code is written in this way already.
Even if sender starts sending emptyObject identifySchemaChange is handling it already. It will compare with the default value.
We should try not to bloat the code as much as possible.

Reply:
If the implementation can handle without having error object, what is the point of returning error?

Reply:
implementation of identifySchemaChange not the Create.
if previousSchema is nil which is current case of Create it will simply return error.